### PR TITLE
Switch to fleetdb and update Go, linter versions

### PR DESCRIPTION
**Pull request by joelrebel**
- Switch to newer GolangCI linter, Go 1.22 and current ConditionOrc
- No breaking changes for end users

Before merge, make sure https://github.com/metal-toolbox/conditionorc/pull/195 is merged and this branch tracks the new Conditionorc release.

### CODE REVIEW

File: pkg/model/model.go [hunk 1]
```diff
@@ -9,30 +9,30 @@ type (
...
 // Config struct holds the mctl configuration parameters
 type Config struct {
 
 	// File is configuration file path
-	File          string
-	Serverservice *ConfigOIDC `mapstructure:"serverservice_api"`
-	Conditions    *ConfigOIDC `mapstructure:"conditions_api"`
-	BomService    *ConfigOIDC `mapstructure:"bomservice_api"`
+	File       string
+	FleetDBAPI *ConfigOIDC `mapstructure:"serverservice_api"` // TODO: implement backwards compatibility and rename.
```
Comment (jakeschuurmans) [inline pkg/model/model.go:22]: Is there a ticket for this TODO?

**Review by jakeschuurmans — DISMISSED** [review]

File: .golangci.yml [hunk 2]
```diff
@@ -27,6 +26,9 @@ linters-settings:
   depguard:
     list-type: blacklist
     packages:
+      # logging is allowed only by logutils.Log, logrus
```
Comment (DoctorVin) [inline .golangci.yml:29]: what's this for? It doesn't seem to actually do anything?

File: .golangci.yml [hunk 3]
```diff
@@ -41,9 +43,10 @@ linters-settings:
       - experimental
     disabled-checks:
       - wrapperFunc
-      - whyNoLint
```
Comment (DoctorVin) [inline .golangci.yml:44]: please don't remove this

File: cmd/list/server.go [hunk 4]
```diff
@@ -138,15 +139,18 @@ func serversTable(servers []*rt.Server, fl *listServerFlags) {
 	table.Render()
 }
 
-func attributeParamsFromFlags(fl *listServerFlags) []ss.AttributeListParams {
-	alp := []ss.AttributeListParams{}
+func attributeParamsFromFlags(fl *listServerFlags) []fleetdbapi.
```
Comment (DoctorVin) [inline cmd/list/server.go:142]: That is an _odd_ linebreak.

**Review by DoctorVin — CHANGES_REQUESTED** [review]

File: pkg/model/model.go (hunk 1 above)
Comment (joelrebel) [reply pkg/model/model.go:22]: Yep, internal reference is FS-1261

File: cmd/list/server.go (hunk 4 above)
Comment (joelrebel) [reply cmd/list/server.go]: good catch, fixed

File: cmd/list/firmware.go [hunk 5]
```diff
@@ -95,4 +97,6 @@ func init() {
 	listFirmware.PersistentFlags().StringSliceVar(&flagsDefinedListFirmware.models, "models", nil, "one or more comma separated models numbers")
 	listFirmware.PersistentFlags().StringVar(&flagsDefinedListFirmware.component, "component", "", "component type")
 	listFirmware.PersistentFlags().StringVar(&flagsDefinedListFirmware.version, "version", "", "version number")
+	// nolint:gomnd // default param is clear as defined
+	listFirmware.PersistentFlags().IntVar(&flagsDefinedListFirmware.limit, "limit", 100, "set result set limit")
```
Comment (DoctorVin) [inline cmd/list/firmware.go:101]: if you are going to support pagination you'll need an `offset` parameter too, yes?

**Review by DoctorVin — APPROVED** [review]
Minor comment about the list firmware API, otherwise LGTM.

**Review by jakeschuurmans — APPROVED** [review]

File: .golangci.yml (hunk 2 above)
Comment (joelrebel) [reply .golangci.yml]: purged

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**